### ISSUE TITLE
Allow setting options through a .clang-tidy file

### ISSUE THREAD

**Issue by chri7325** [Caphyon/clang-power-tools#156]
For our project, we have all of our options set through a .clang-tidy file in the root of our codebase.  This file controls all options that we use and makes it easy to configure as we only use a subset of extensions.  Currently, I don't see a way to allow the clang-tidy option to have a config=file option.  By default, clang-tidy will recurse upwards from the parent directory and if any .clang-tidy files are found with configurations, it will use that.  If this could be set in the plugin, then I could have my checks, WarningsAsErrors, HeaderFilterRegex, AnalyzeTemporaryDtors and FormatStyle all set without asking other developers to manually set them up.

**Comment by cpp-red-lion** [comment]
Hi, 

This a very good suggestion. Thank you.
We'll implement it in the next version of the VS extension.

I'll let you know when it's ready for a test-drive.

**Comment by chri7325** [comment]
@cpp-red-lion Thanks!  Please let me know if there is a branch you want me to test or anything and I'll be very happy to give it a go.

**Comment by cpp-red-lion** [comment]
The changes in the CI/automation PowerShell script already landed in `master` branch.
For the VS extension, it will take a little more time to push the UI changes in the next version. (max 1 day)

In the mean time, if you want to test-drive the PS script:

- Open **PowerShell** console in _your project_ root directory.
- Fetch Clang Power Tools PS files from GitHub repo:

`Invoke-WebRequest -Uri "https://raw.githubusercontent.com/Caphyon/clang-power-tools/master/ClangPowerTools/ClangPowerTools/sample-clang-build.ps1" -OutFile ".\run-clang.ps1"`

`Invoke-WebRequest -Uri "https://raw.githubusercontent.com/Caphyon/clang-power-tools/master/ClangPowerTools/ClangPowerTools/clang-build.ps1" -OutFile ".\clang-build.ps1"`

- (optionally) customize `run-clang.ps1` bootstrapper script with your desired configs (constants)
- run clang build:

`.\run-clang.ps1 -tidy-fix .clang-tidy -proj YourProj -file Source.cpp`

( `-file`, `-proj`, `-file-ignore`, `-proj-ignore` are all optional )

**Comment by hero101111** [comment]
We implemented this feature in version `2.3` we just released.

**Comment by chri7325** [comment]
@hero101111 Thank you!  I have downloaded it and I'm giving it a try.  I see the options there now but now I'm getting failures when running clang-tidy.exe.  I'm guessing this is due to the other enhancement of using pch files with clang?  The failure generated is that the pch object was created with a different tag of clang.  Would the fix be also using our own clang.exe so that the versions match?  Or is there a way to turn that enhancement off?

**Comment by hero101111** [comment]
I don't follow what's happening here. Why should there be two versions of clang used?
Please give more info on this.

Regarding a workaround, unfortunately there's no official way to disable PCH usage in the tidy workflow.
However, you can manually edit the `clang-build.ps1` inside the VS Extension directory and comment the code that passes the PCH to clang-tidy. 
- Open `%localappdata%` in Windows Explorer and search for `clang-build.ps1`
- Open script in your editor of choice
- Locate the function `Get-TidyCallArguments`
- Comment the `If` statement for `$pchFilePath` (should be at line 1601)
- Save script & Restart Visual Studio

**Comment by chri7325** [comment]
@hero101111 I was able to workaround this problem.  The issue here is that I had LLVM/bin on my path to use the tools.  I have my own version of clang-tidy.exe with my custom extension on my PATH as well that I wanted to be used instead of the installer's version.  However, I didn't have the same version of clang.exe.  With the pch enhancements that were made, it was using the installer's clang.exe to build the pch object (LLVM version 5.0.0) and then tried analyzing the code with my clang-tidy (LLVM version 5.0.0git-55cce4b87ee).   Because of this, errors were generated immediately.  I have uninstalled LLVM and am now using the clang.exe that I built along with clang-tidy.exe.  Once I did that, the versions matched and it runs again.
